spec.md

```diff
@@ -522,6 +535,165 @@ the module.  The `c`{.firrtl} sub-field contained in the `b`{.firrtl} sub-field
 flows into the module, and the `d`{.firrtl} sub-field contained in the
 `b`{.firrtl} sub-field flows out of the module.
 
+
+## Reference Types
+
+References can be exported from a module for indirect access elsewhere,
+and are captured using values of reference type.
+
+For use in cross-module references (XMR's), a reference to a probe of the
+circuit component is used.  See [@sec:probes] for details.
+
+Using reference-type ports, modules may expose internals for reading and
+forcing without routing wires out of the design.
+
+This is often useful for testing and verification, where reference types allow
+reads of the entities to be explicitly exported without hard-coding their place
+in the design.  Instead, by using references, a testbench module may express
+accesses to the internals which will resolve to the appropriate target language
+construct by the compiler (e.g., hierarchical reference).
+
+Reference ports are not expected to be synthesizable or representable in the
+target language and are omitted in the compiled design; they only exist
+at the FIRRTL level.
+
+Reference-type ports are statically routed through the design using
+the `forward`{.firrtl} and `export`{.firrtl} statements.
+
+There are two reference types, `Probe`{.firrtl} and `RWProbe`{.firrtl},
+described below.  These are used for indirect access to probes of the data
+underlying circuit constructs they originate from, captured using
+`probe`{.firrtl} expressions (see [@sec:probes]).
+
+`Probe`{.firrtl} types are read-only, and `RWProbe`{.firrtl} may be used with
+`force`{.firrtl} and related statements.
+Prefer the former as much as possible, as read-only probes impose fewer
+limitations and are more amenable to optimization.
+
+References must always be able to be statically traced to their target, or to
+an external module's output reference.
+
+### Probe Types
+
+Probe types are reference types used to access circuit elements' data remotely.
+
+There are two probe types: `Probe`{.firrtl} and `RWProbe`{.firrtl}.
+`RWProbe`{.firrtl} is a `Probe`{.firrtl} type, but not the other way around.
```

Comment:
Why is this inheritance relationship necessary? It seems overly complicated?

Reply:
This is really just saying that you can `read` from a `RWProbe`, but you can't `force`/`release` a `Probe`.

It may be simpler to not frame this as "inheritance" and to just state it in the operations. A `read` can use a `RWProbe` or a `Probe`. A `force` can use a `RWProbe`.

Reply:
FWIW, it also is the means by which you're allowed to target a `Probe`-type reference with a `RWProbe` source.  You can drop the ability to be used with `force` if you'd like, is the idea.

Thank you both, will see about how this might be better clarified.  Sub-type was supposed to help folks come to expected conclusions re:substitutability, aka meant to be making it clearer/simpler not more complicated :smile: .  Whoops.